packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/stepfunctions/start-execution.ts

```diff
@@ -42,6 +42,15 @@ export interface StepFunctionsStartExecutionProps extends sfn.TaskStateBaseProps
    * @default - false
    */
   readonly associateWithParent?: boolean;
+
+  /**
+   * The Step Functions state machine ARN to start the execution on.
+   * This allows you to specify a state machine version or alias ARN.
+   * If specified, it overrides the ARN of the stateMachine.
+   *
+   * @default - The ARN of StepFunctionsStartExecutionProps.stateMachine is used
+   */
+  readonly stateMachineArn?: string;
```

Comment:
Why do this, instead writing:

```ts
new StartExecution({
  stateMachine: StateMachine.fromStateMachineArn(this, 'SM', 'arn:aws:...'),
});
```

Like we do for every other construct?

Reply:
I guess that works from looking at the implementation. However, StateMachineVersion and StateMachineAlias are different constructs. The ARN of an alias (e.g. `arn:aws:....:stateMachine:MyStateMachineName:MyAlias`) is not a state machine ARN

https://github.com/aws/aws-cdk/blob/27b720135e30d58732997a5a5e172daaa9453381/packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts#L166-L168

If that's the preferred approach, it needs to be documented.